src/main/kotlin/uk/gov/dluhc/printapi/service/SftpService.kt

```diff
@@ -72,9 +72,17 @@ class SftpService(
         return responseString!!
     }
 
+    @Throws(IOException::class)
     fun removeFileFromOutBoundDirectory(directory: String, fileName: String): Boolean {
         logger.info { "Removing processed file [$fileName] from directory [$directory]" }
-        return sftpOutboundTemplate.remove(createFileNamePath(directory, fileName))
+        try {
```

Comment:
Do you need to do this? (catching the exception and either throwing as an IOException, or wrapping it in one)
Vishal didnt need to do it in his method in this class?

AFAIK the SFTP Template methods only throw MessagingException, and AFAIK MessagingException is already an IOException ??

Reply:
@NathanRussellValtech, I have tried it the way Vishal did it by just adding the annotation, but it didn't work. 
Also checked if `MessagingException` is an `IOException`, but it doesn't look like it is. Hence, I went with the current implementation.
<pre>
package org.springframework.messaging;
public class MessagingException extends NestedRuntimeException 

public abstract class NestedRuntimeException extends RuntimeException
</pre>

Reply:
You are 100% right! and Vishal's implementation is wrong!
The SFTP Template definitely only throws MessagingException, so can you change your impl here to:
```
catch (ex: MessagingException) {
    throw IOException(ex)
}
```
The comments I left on your tests would then still stand, in that we'd need the exception to be thrown to be a MessagingException

Dont worry about Vishal's impl here (which is wrong!) - I'll fix it later and do a separate PR for it 👍

Reply:
Just to confirm your suggestion @NathanRussellValtech , do you want me to change this:

<pre>
    fun removeFileFromOutBoundDirectory(directory: String, fileName: String): Boolean {
        logger.info { "Removing processed file [$fileName] from directory [$directory]" }
        try {
            return sftpOutboundTemplate.remove(createFileNamePath(directory, fileName))
        } catch (ex: Exception) {
            when (ex.cause) {
                is IOException -> throw ex.cause as IOException
                else -> throw IOException(ex)
            }
        }
    }
</pre>

to this:
<pre>
    fun removeFileFromOutBoundDirectory(directory: String, fileName: String): Boolean {
        logger.info { "Removing processed file [$fileName] from directory [$directory]" }
        try {
            return sftpOutboundTemplate.remove(createFileNamePath(directory, fileName))
        } catch (ex: MessagingException) {
            throw IOException(ex)
        }
    }
</pre>

If the answer is yes, then `SftpService` will still be throwing `IOException` and in the `PrintResponseProcessingService` we will still need to mock it to throw `IOException` instead of `MessagingException`.

Maybe I am missing something with your comment.

Reply:
The answer is Yes; and yes, you are right about the tests! 😁 👍